lib/theme-api-client.js

```diff
@@ -77,38 +65,31 @@ async function deleteThemeById({ themeId, accessToken, apiHost, storeHash }) {
  * @returns {Promise<object>}
  */
 async function getJob({ accessToken, apiHost, storeHash, jobId, resultFilter }) {
-    const reqOpts = {
-        headers: {
-            'cache-control': 'no-cache',
-            'x-auth-client': 'stencil-cli',
-            'x-auth-token': accessToken,
-        },
-    };
-    const reqUrl = `${apiHost}/stores/${storeHash}/v3/themes/jobs/${jobId}`;
     let response;
     let payload;
-
     try {
-        response = await fetch(reqUrl, reqOpts);
-        payload = await response.json();
+        response = await sendApiRequest({
+            url: `${apiHost}/stores/${storeHash}/v3/themes/jobs/${jobId}`,
+            headers: {
+                'cache-control': 'no-cache',
+                'x-auth-client': 'stencil-cli',
+                'x-auth-token': accessToken,
```

Comment:
do we need to add `content-type` header here ? I am wondering if we should remove the headers outside and keep it generic to all calls.

Reply:
@junedkazi 
It's a GET request, so we aren’t sending any data here (no content, so no content-type).

Regarding moving the headers inside `sendApiRequest` - it's a good idea, but there are no headers that we send in all our requests. Though, we can move 
```
{
    'x-auth-client': 'stencil-cli',
    'stencil-cli': PACKAGE_INFO.version,
    'stencil-version': PACKAGE_INFO.config.stencil_version,
}
```
since we are sending them in the majority of the requests and they won’t hinder us if we send them when not required.

Reply:
ya we should consolidate the generic ones in a single place and avoid duplication

Reply:
Updated in the new commit